src/sinks/mod.rs

```diff
@@ -87,3 +88,29 @@ pub enum HealthcheckError {
     #[snafu(display("Unexpected status: {}", status))]
     UnexpectedStatus { status: ::http::StatusCode },
 }
+
+impl VectorSink {
+    pub async fn run<S>(self, input: S) -> Result<(), ()>
+    where
+        S: futures01::Stream<Item = Event, Error = ()>,
+    {
+        match self {
+            Self::Futures01Sink(sink) => input.forward(sink).compat().map_ok(|_| ()).await,
+        }
+    }
+
+    pub fn into_futures01sink(
```

Comment:
Is this method just for tests? I can see it being useful for migration, but we will probably not want to keep it around.

Reply:
Not only for tests, but also as bypass for `StreamSink`.
Actually, I can use `run` method in tests instead converting to `Sink`, this will be more correct way. Will change it right now.

Reply:
Updated. Now used for `StreamSink`(sematext, file, console) and in 1 test (tcp_stream_detects_disconnect in socket sink).